source/common/grpc/async_client_impl.h

```diff
@@ -49,8 +49,22 @@ class AsyncClientImpl final : public RawAsyncClient {
   friend class AsyncStreamImpl;
 };
 
+// CallbackHolder is a mixin class that holds a callback that has a lifetime
+// as long as the object that holds it.
+class CallbackHolder {
+public:
+  typedef std::unique_ptr<RawAsyncStreamCallbacks> CallbackPtr;
+  CallbackPtr& callbackHolder() { return callback_; }
+  void setCallbackHolder(CallbackPtr&& callback) { callback_ = std::move(callback); }
```

Comment:
only comment here would be that the mixin is `CallbackHolder`, but the setter and getter deal with the callback that is being held, not the holder itself. would `getHeldCallback` and `setHeldCallback` better represent the intention?

Reply:
Good point. Updated.